engine/execution/ingestion/engine.go

```diff
@@ -180,6 +186,116 @@ func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 	}
 }
 
+// on nodes startup, we need to load all the unexecuted blocks to the execution queues.
+func (e *Engine) loadAllFinalizedAndUnexecutedBlocks() error {
+	// // get finalized height
+	header, err := e.state.Final().Head()
+	if err != nil {
+		return fmt.Errorf("could not get finalized block: %w", err)
+	}
+
+	finalizedHeight := header.Height
+	futureHeight := uint64(8655590)
+
+	// get the last executed height
+	lastExecutedHeight, _, err := e.execState.GetHighestExecutedBlockID(e.unit.Ctx())
+	if err != nil {
+		return fmt.Errorf("could not get last executed block: %w", err)
+	}
+
+	unexecuted := int64(finalizedHeight) - int64(lastExecutedHeight)
+
+	e.log.Info().
+		Int64("count", unexecuted).
+		Uint64("last_executed_height", lastExecutedHeight).
+		Uint64("last_finalized_height", finalizedHeight).
+		Msg("reloading finalized and unexecuted blocks to execution queues...")
+
+	// log the number of unexecuted blocks
+	if unexecuted <= 0 {
+		return nil
+	}
+
+	count := 0
+	for height := lastExecutedHeight + 1; height <= futureHeight; height++ {
```

Comment:
```suggestion
	for height := lastExecutedHeight; height <= futureHeight; height++ {
```
There is no hard requesting an already executed block, and it's better to overlap, as it solves some current issues with the lastExecutedHeight not being saved idempotently with the rest of the results